### DISCUSSION TITLE
configure-ssh-keys-for-multiple-github-accounts

### DISCUSSION THREAD

**Discussion by giscus** [stevenharman/stevenharman.github.io#21] [Announcements]
# configure-ssh-keys-for-multiple-github-accounts

Managing different SSH Keys for different Hosts is well-understood. But different keys for the same host (e.g., github.com), based on which Organization we'r...

https://stevenharman.net/configure-ssh-keys-for-multiple-github-accounts?ref=dailydev

**Comment by AWRootbeard** [comment]
Great solution.  I think there's a small bug in your example code.  The insteadOf is replacing "git@github.com:planet-express" with "github-plnx:planet-express" (without the "git@").  This threw me off for a bit, but once I made them match it started working for me.

**Reply by stevenharman** [reply]
Hello @AWRootbeard!

For some reason I didn't get a notification about your comment. I'm not sure I understand what change I need to make as I am using code an `insteadOf` just like the examples in my day to today, and it's working as expected. Perhaps I'm just failing to "see" the typo you found?

If you could propose a change to [the article itself](https://github.com/stevenharman/stevenharman.github.io/blob/main/_posts/2024-08-21-configure-ssh-keys-for-multiple-github-accounts.md), maybe that will help me understand/see the problem?

Thanks.

**Reply by stevenharman** [reply]
Wait, now I see. This is actually intended. If you look at the `~/.ssh/config` you'll see:

```sshconfig
Host github-plnx
  HostName github.com
  User git
```

I explain in the article the following:

> Then the SSH config matches the host `github-plnx`, and using that matched config, replaces `github-plnx` with the actual user (`git`) and host name (`github.com`).

So it could be that your `~/.ssh/config` was missing the `User git` part?
